File: Solutions/0495/0495.js
var findPoisonedDuration = function(timeSeries, duration) {
    if (duration === 0)
           return 0;

       let ans = 0;

       for (let i = 0; i + 1 < timeSeries.length; ++i)
           ans += Math.min(timeSeries[i + 1] - timeSeries[i], duration);

       return ans + duration;
};

File: Solutions/0495/0495.go
func findPoisonedDuration(timeSeries []int, duration int) int {
	if duration == 0 {
		return 0
	}

	ans := 0

	for i := 0; i+1 < len(timeSeries); i++ {
		ans += min(timeSeries[i+1]-timeSeries[i], duration)
	}

	return ans + duration
}